index.html

```diff
@@ -334,6 +337,47 @@ <h2>
             {{Gamepad/id}} string.
           </p>
         </dd>
+        <dt>
+          <dfn>vendorId</dfn> attribute
+        </dt>
+        <dd>
+          <p>
+            The 16-bit identifier assigned to the device vendor by the USB
+            Implementers Forum or Bluetooth Special Interest Group, or `null`
+            if the gamepad has no vendor identifier or it is not available.
+          </p>
+          <p>
+            The [=user agent=] MAY set {{Gamepad/vendorId}} and
+            {{Gamepad/productId}} to `null` to mitigate the risk of exposing
+            device identifiers. If one of {{Gamepad/vendorId}} or
+            {{Gamepad/productId}} is `null` then both MUST be `null`.
+          </p>
+        </dd>
+        <dt>
+          <dfn>productId</dfn> attribute
+        </dt>
+        <dd>
+          The product identifier assigned by the device vendor, or `null` if
+          the gamepad has no product identifier or it is not available.
+        </dd>
+        <dt>
+          <dfn>name</dfn> attribute
+        </dt>
+        <dd>
+          <p>
+            A human-readable identifier for the gamepad. The exact content of
+            the {{Gamepad/name}} string is left unspecified. The [=user agent=]
+            MAY use a product name string provided by the device firmware. The
+            {{Gamepad/name}} MUST NOT be the empty string and MUST NOT include
+            unique identifiers like serial numbers or Bluetooth device
```

Comment:
Maybe make it more clear that this means the vendorId and productId?

Reply:
I don't consider vendor/product IDs unique identifiers since they typically have the same values for all instances of the same product. The language here was intended to stop implementers from doing something like appending the USB serial number string descriptor to the name attribute as a way to provide a unique identifier. We want to give the implementation some leeway to pick a good name but we shouldn't allow this attribute to be used to hold a unique identifier since that would violate our assumptions around fingerprintability.

Including vendor/product IDs shouldn't make the name string any more useful as a unique identifier since we expect these IDs will correlate strongly with the product name, but I'd argue it still doesn't satisfy the spec because vendor and product IDs aren't really human-readable.

Reply:
ok, yeah, that makes sense. I was more thinking about discouraging implementers from putting weird hex numbers etc. that would then end up being presented to users (which would be meaningless for most users). The description does make it clear that that this is supposed to be an end-user facing string.